project-base/storefront/components/Pages/Customer/NotAddedProductsPopup.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button } from 'components/Forms/Button/Button';
+import { Popup } from 'components/Layout/Popup/Popup';
+import { getInternationalizedStaticUrls } from 'helpers/localization/getInternationalizedStaticUrls';
+import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import { useRouter } from 'next/router';
+import { Dispatch, SetStateAction } from 'react';
+
+type NotAddedProductsPopupProps = {
+    notAddedProductNames: string[];
+    setNotAddedProductNames: Dispatch<SetStateAction<string[] | undefined>>;
+};
+
+export const NotAddedProductsPopup: FC<NotAddedProductsPopupProps> = ({
+    notAddedProductNames,
+    setNotAddedProductNames,
+}) => {
+    const t = useTypedTranslationFunction();
+    const router = useRouter();
+    const { url } = useDomainConfig();
+    const [cartUrl] = getInternationalizedStaticUrls(['/cart'], url);
+
+    return (
+        <Popup onCloseCallback={() => setNotAddedProductNames(undefined)}>
```

Comment:
Same here. No need to pass whole setter function. Enough in this case would be pass simple void callback () => void and have setter function in the parent.

Then we can call this callback function just onClose or similar.

Reply:
I hope I understood your request correctly... If not, let me know and I will fix it :slightly_smiling_face: